### ISSUE TITLE
 MLS 3.5 RC1: MLS.pdf p194 misses line breaks

### ISSUE THREAD

**Issue by beutlich** [modelica/ModelicaSpecification#2796]
Page 194 (section 13.5) of the current MLS 3.5 RC1 from https://specification.modelica.org/maint/3.5/MLS.pdf has some long lines, making it not visually appealing. There might be more of such occurrrences since I only detected these ones randomly.

![grafik](https://user-images.githubusercontent.com/14896695/103700836-52115d80-4fa5-11eb-85c6-efc8f7a01504.png)

**Comment by beutlich** [comment]
Furthermore: The explicit version reference "3.2.1" of Modelica 3.2.1 should be removed, too.

**Comment by henrikt-ma** [comment]
> Page 194 (section 13.5) of the current MLS 3.5 RC1 from https://specification.modelica.org/maint/3.5/MLS.pdf has some long lines, making it not visually appealing. There might be more of such occurrrences since I only detected these ones randomly.

It is true that it looks bad, but we must also weigh maintenance cost against the looks of the increasingly dated PDF format.  It is definitely possible to go through the entire PDF document after all other changes have been merged, and work out all line and page breaks, but fixing it is both costly and sometimes requires tricks that make the document harder to maintain in the long run.

**Comment by beutlich** [comment]
I'd prefer to set the explicit hypenation to improve the overall layout. I am aware that this comes with some extra cost, but you only need to do it once for the few occurrences.

**Comment by henrikt-ma** [comment]
> I'd prefer to set the explicit hypenation to improve the overall layout. I am aware that this comes with some extra cost, but you only need to do it once for the few occurrences.

The problem is that it doesn't really end after you've made a paragraph pretty.  When the paragraph gets edited in the future, the things done now to make it look good might actually destroy nice line breaking of the modified text.  One option for taking line breaking seriously would be to only do it on release branches, and make sure that _master_ remains free of these fragile line breaking tweaks.  @HansOlsson, how would that sound to you?

**Comment by HansOlsson** [comment]
I tried to look through everything:
Page 90 also has a bad case for BaseProperties; page 106 for startExpression; page 183 for ExternalFunction, page 123 for shiftSample; page 247 for tryme, page 258 EllipseClosure, page 265 connectorSizing, page 267 STRING.
And page 272 has a section-heading that is too long "Annotations for Access Control to Protect Intellectual Property"
Page 281 has HASE'08 as too long entry in bibliography.

To me there is only case that really stands out and that is the 2nd case in the screenshot above. Most of the others are below some "acceptable threshold" or there is no obvious margin-reference so you don't notice it directly.

**Comment by HansOlsson** [comment]
> > I'd prefer to set the explicit hypenation to improve the overall layout. I am aware that this comes with some extra cost, but you only need to do it once for the few occurrences.
> 
> The problem is that it doesn't really end after you've made a paragraph pretty. When the paragraph gets edited in the future, the things done now to make it look good might actually destroy nice line breaking of the modified text. One option for taking line breaking seriously would be to only do it on release branches, and make sure that _master_ remains free of these fragile line breaking tweaks. @HansOlsson, how would that sound to you?

I understand this concern, but in many cases I wouldn't expect that they are modified every release (in particular not the bibliography) so I would more say that we between releases preserve existing good line tweaks, but don't add new ones until close to the release - and if an existing one becomes broken we can remove it.

**Comment by HansOlsson** [comment]
> Furthermore: The explicit version reference "3.2.1" of Modelica 3.2.1 should be removed, too.

I would more say reworked. Showing that references work for packages with version as part of package name seems good, and having an actual version seems better for that, but it makes sense to also have the other case and actually have the other case as default.

**Comment by henrikt-ma** [comment]
> > Furthermore: The explicit version reference "3.2.1" of Modelica 3.2.1 should be removed, too.
> 
> I would more say reworked. Showing that references work for packages with version as part of package name seems good, and having an actual version seems better for that, but it makes sense to also have the other case and actually have the other case as default.

But are they really different cases?  The version isn't a part of the package name, it's just a library stored in a directory reflecting the version of the package it contains.  As far as I understand it is completely up to tools and their users to organize how libraries are stored, and the closest thing I can think of in the specification is the discussion of `MODELICAPATH`, where there is no sign of version numbers being present in directory names to help the search for a specific version of a library.

For what it's worth, in the specific case of the MSL, I'd actually put the version number in the name of the repository's root directory, so it doesn't even look right to me to have a version number attached to the directory name _Modelica_.

In summary, I like @beutlich's suggestion to simply drop the version number in this case.

**Comment by henrikt-ma** [comment]
> I understand this concern, but in many cases I wouldn't expect that they are modified every release (in particular not the bibliography) so I would more say that we between releases preserve existing good line tweaks, but don't add new ones until close to the release - and if an existing one becomes broken we can remove it.

OK, let's do it.  I guess you kind of self-assigned when going through the document to find the things in need of attention?

**Comment by HansOlsson** [comment]
> But are they really different cases? The version isn't a part of the package name, it's just a library stored in a directory reflecting the version of the package it contains. As far as I understand it is completely up to tools and their users to organize how libraries are stored, and the closest thing I can think of in the specification is the discussion of `MODELICAPATH`, where there is no sign of version numbers being present in directory names to help the search for a specific version of a library.

It is actually defined - but hidden in the wrong chapter (well, there is a reference to it):
https://specification.modelica.org/maint/3.5/annotations.html#mapping-of-versions-to-file-system

**Comment by henrikt-ma** [comment]
> It is actually defined - but hidden in the wrong chapter (well, there is a reference to it):
> https://specification.modelica.org/maint/3.5/annotations.html#mapping-of-versions-to-file-system
> 
> I'm not sure if it is intended as a help - or more as way of allowing multiple versions of a library to exist in the same folder.

Oh.  Considering that this doesn't even seem like a good approach for someone installing the MSL on their `MODELICAPATH`, I believe it's time to tear this up.  Can't we investigate to what degree `MODELICAPATH` is actually used as described by tools, and if it turns out not to be used as described by most tools, simply remove it completely?

**Comment by gkurzbach** [comment]
In SimulationX we implemented MODELICAPATH as described in chapter 13.3 as an entry in the windows registry. In the list are not the directories where the package.mo file live, but one level above (as described there). And we implemented the mapping of versions to file system as described in 18.3.3. But in chapter 18.3.3. there is the sentence: 

A top-level class, IDENT, with version VERSION-NUMBER can be stored in one of the following ways **in a directory given in the MODELICAPATH**:

Example: Modelica 2.1/package.mo

This "in a directory given in the MODELICAPATH" could be slidely misleading because this can be interpreted as "the directory 'Modelica 2.1' has to be in the MODELICAPATH" (which is wrong) or "there exist a directory in MODELICAPATH which contains 'Modelica 2.1'". But after reading the sentence the third time, the first variant becomes much less likely.

**Comment by henrikt-ma** [comment]
> In SimulationX we implemented MODELICAPATH as described in chapter 13.3 as an entry in the windows registry.

System Modeler does not read any `MODELICAPATH` environment variable, and according to the OpenModelica documentation, they don't read it either:
https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/faq.html

However, the specification is making it clear that `MODELICAPATH` isn't required to be an environment variable, but that it's completely up to tools to define how to store the list (like, SimulationX does, for example).  However, this also means that `MODELICAPATH` adds very little value for tool independent organization of libraries, and it seems mostly like an irrelevant specification of detail to say exactly how to organize libraries on the tool-dependent `MODELICAPATH`.

> Example: Modelica 2.1/package.mo

While it is possible to put a distribution of a MSL release (available from https://github.com/modelica/ModelicaStandardLibrary/releases) on the `MODELICAPATH`, it is not possible to have a repository clone on the `MODELICAPATH` since having version numbers included in the subdirectory names would be madness.  Other libraries with a similar organization as the MSL will suffer from the same problem, and it doesn't exactly help that we don't have a system for giving build descriptions that can be used to compile distribution products like what there is for the MSL.

**Comment by HansOlsson** [comment]
In Dymola (both standalone and as part of 3D Experience platform) we use MODELICAPATH, and if you set the environment variable inside Dymola that changes the library search.

However, we don't set the windows-wide MODELICAPATH (since that isn't nice to other programs; including other versions), and we thus amend it inside Dymola in a slightly complicated way (described somewhere in the manual) to include the libraries that are part of the current Dymola installation (as anything else would be confusing).

**Comment by HansOlsson** [comment]
> This "in a directory given in the MODELICAPATH" could be slidely misleading because this can be interpreted as "the directory 'Modelica 2.1' has to be in the MODELICAPATH" (which is wrong) or "there exist a directory in MODELICAPATH which contains 'Modelica 2.1'". But after reading the sentence the third time, the first variant becomes much less likely.

I agree that it should be clarified like that.

**Comment by HansOlsson** [comment]
> > In SimulationX we implemented MODELICAPATH as described in chapter 13.3 as an entry in the windows registry.
> 
> System Modeler does not read any `MODELICAPATH` environment variable, and according to the OpenModelica documentation, they don't read it either:
> https://www.openmodelica.org/doc/OpenModelicaUsersGuide/latest/faq.html

However, I'm pretty sure that some version did use it - and also did set the global environment variable.
